tasks/main.yml

```diff
@@ -41,15 +41,10 @@
   when: ansible_os_family != "Darwin"
   register: nix_directory
 
-- name: Set should_install_nix fact
-  ansible.builtin.set_fact:
-    should_install_nix: true
-    when: >
-      (ansible_os_family != "Darwin" and nix_directory is changed) or
-      (ansible_os_family == "Darwin" and not nix_directory_mac.stat.exists)
-
 - name: Install Nix
-  when: nix_directory is changed  # noqa no-handler
```

Comment:
Hmm, actually, we should just use `when: should_install_nix`. I'll open up a new PR to also spare you having to sign our CLA.

Reply:
Ah, actually, the problem here was the indentation level of the `when` condition. :sweat_smile: :facepalm: